src/System.ComponentModel.TypeConverter/src/System/ComponentModel/MemberDescriptor.cs

```diff
@@ -421,18 +422,24 @@ private void FilterAttributesIfNeeded()
                 }
                 else
                 {
-                    list = new ArrayList(_attributes);
+                    list = new List<Attribute>(_attributes);
                 }
 
-                Hashtable hash = new Hashtable(list.Count);
+                var set = new HashSet<object>();
 
-                foreach (Attribute attr in list)
+                for (int i = 0; i < list.Count;)
                 {
-                    hash[attr.GetTypeId()] = attr;
+                    if (set.Add(list[i].GetTypeId()))
+                    {
+                        ++i;
+                    }
+                    else
+                    {
+                        list.RemoveAt(i);
+                    }
                 }
```

Comment:
@stephentoub Is there guidance in .NET Core about using LINQ? Using `list.Distinct().ToArray()` would be a lot more readable, but I've seen some debate about using LINQ due to some perf concerns.